buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1779,6 +1779,33 @@ class BeamModulePlugin implements Plugin<Project> {
         cleanupTask.mustRunAfter pythonTask
         config.cleanupJobServer.mustRunAfter pythonTask
       }
+      // Task for running testcases in Python SDK
+      def testOpts = [
+        "--attr=UsesSqlExpansionService"
+      ]
+      def pipelineOpts = [
+        "--runner=PortableRunner",
+        "--environment_cache_millis=10000",
+        "--job_endpoint=${config.jobEndpoint}"
+      ]
+      def beamPythonTestPipelineOptions = [
+        "pipeline_opts": pipelineOpts,
+        "test_opts": testOpts,
+        "suite": "xlangSqlValidateRunner"
+      ]
+      def cmdArgs = project.project(':sdks:python').mapToArgString(beamPythonTestPipelineOptions)
+      def pythonSqlTask = project.tasks.create(name: config.name+"PythonUsingSql", type: Exec) {
```

Comment:
You will run the same test twice if you put the task here without using `sdk` and `port`. It will run twice against Python expansion-service and Java expansion-service but `pythonSqlTask` uses neither of them.

I would instead add `:sdks:java:extensions:sql` dependency to existing test expansion service and utilize environment variables `EXPANSION_JAR` and `EXPANSION_PORT` like `pythonTask` above. Since external sql transform is only available on Java expansion service, you should handle the exception like https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/external/xlang_parquetio_test.py#L71

Reply:
If it's outside of the each block then it should be fine. The advantage of using existing testing expansion service here is you don't need to start and stop another expansion service inside the test code. The expansion service starts and finishes at the beginning and the end of the whole test suite. You can just pass the jar name and the port number by environment variables. Still there's no harm in launching your own expansion service though.